routers/private/hook_pre_receive.go

```diff
@@ -124,8 +124,13 @@ func HookPreReceive(ctx *gitea_context.PrivateContext) {
 			preReceiveBranch(ourCtx, oldCommitID, newCommitID, refFullName)
 		case strings.HasPrefix(refFullName, git.TagPrefix):
 			preReceiveTag(ourCtx, oldCommitID, newCommitID, refFullName)
-		case git.SupportProcReceive && strings.HasPrefix(refFullName, git.PullRequestPrefix):
+		case git.SupportProcReceive && strings.HasPrefix(refFullName, git.AgitPullPrefix):
 			preReceivePullRequest(ourCtx, oldCommitID, newCommitID, refFullName)
+		case strings.HasPrefix(refFullName, git.PullPrefix) && newCommitID == git.EmptySHA:
```

Comment:
If the pull request is closed, I think it should be allowed to be deleted.

Reply:
We should probably do that within the UI.